src/middlewared/middlewared/plugins/account.py

```diff
@@ -491,39 +511,42 @@ def do_home_copy():
                 {'prefix': 'bsdgrpmember_'}
             )
 
-    async def __update_sshpubkey(self, user, group):
+    async def __update_sshpubkey(self, homedir, user, group):
         if 'sshpubkey' not in user:
             return
-        keysfile = f'{user["home"]}/.ssh/authorized_keys'
-        pubkey = user.get('sshpubkey')
-        if pubkey is None:
-            if os.path.exists(keysfile):
-                try:
-                    os.unlink(keysfile)
-                except OSError:
-                    pass
-        else:
-            oldpubkey = ''
+        if not os.path.isdir(homedir):
+            return
+
+        sshpath = f'{homedir}/.ssh'
+        keysfile = f'{sshpath}/authorized_keys'
+
+        pubkey = user.get('sshpubkey', '').strip()
```

Comment:
@b-a-t what if sshpubkey is None? I believe thats a possibility, so you would try to strip() NoneType here.

Reply:
That's why the second default parameter is `''`, so we get empty string instead.

Reply:
I think you have a misunderstanding of how get works:

```
>>> user = {'sshpubkey': None}
>>> pubkey = user.get('sshpubkey', '').strip()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'strip'```

Reply:
It's more like I didn't expect that someone would be so evil to pass `None` as a value :)